module/Finna/src/Finna/RecordDriver/SolrMarc.php

```diff
@@ -165,6 +166,33 @@ public function getAccessRestrictionsType($language)
         return false;
     }
 
+    /**
+     * Get all IIIF manifests.
+     *
+     * Finds all 'u' subfields in field 856 with 'q' matching a IIIF
+     * Presentation API content type.
+     *
+     * @return array
+     */
+    public function getAllIiifManifests()
+    {
+        $reader = $this->getMarcReader();
+        $field856 = $reader->getFields('856', ['q', 'u']) ?? [];
+        $manifests = [];
+        foreach ($field856 as $field) {
+            $u = $reader->getSubfield($field, 'u');
+            $q = $reader->getSubfield($field, 'q');
+            if (
+                !empty($u) &&
+                !empty($q) &&
+                $this->isIiifPresentationManifest($q)
```

Comment:
$u on aina stringi, niin riittää tsekata noin (ja jättää q hakematta, jos u sattuisi puuttumaan):
```suggestion
            if (
                $u
                && $this->isIiifPresentationManifest($reader->getSubfield($field, 'q'))
```